src/core/namespace.c

```diff
@@ -2157,7 +2157,10 @@ int setup_namespace(
                                 goto finish;
                 }
 
-                if (ns_info->protect_hostname) {
+                /* If proc is mounted with subset=pid then neither of the two
+                 * paths will exist, i.e. they are implicitly protected by the
+                 * mount option. */
+                if (ns_info->protect_hostname && !ns_info->proc_subset == PROC_SUBSET_PID) {
```

Comment:
Oh, one thing I should check is how this is handled if the kernel doesn't support `subset=pid`, i.e. do we at that point already now that `PROC_SUBSET_PID` won't work and will have unset `protect_subset`. If not we need to tweak the patch.

Reply:
see other comment, we gracefully fall back in that case

Reply:
fixed